File: android/src/main/kotlin/com/emarsys/emarsys_sdk/command/predict/TrackItemViewCommand.kt
package com.emarsys.emarsys_sdk.command.predict

import com.emarsys.Emarsys
import com.emarsys.emarsys_sdk.command.EmarsysCommand
import com.emarsys.emarsys_sdk.command.ResultCallback

class TrackItemViewCommand : EmarsysCommand {
    override fun execute(parameters: Map<String, Any?>?, resultCallback: ResultCallback) {
        val itemId: String? = parameters?.get("itemId") as String?
        itemId ?: throw IllegalArgumentException("itemId should not be null!")

        Emarsys.predict.trackItemView(itemId)

        resultCallback.invoke(null, null)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return true
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}

File: ios/Classes/Wrapper/Commands/Predict/TrackItemViewCommand.swift
//
// Created by Emarsys
//

import EmarsysSDK

class TrackItemViewCommand: EmarsysCommandProtocol {
    func execute(arguments: [String: Any]?, resultCallback: @escaping ResultCallback) {
        guard let itemView = arguments?["itemView"] as? String else {
            resultCallback(["error": "Invalid itemView argument"])
            return
        }
        
        Emarsys.predict.trackItem(itemId : itemView)

        resultCallback(["success": true])
    }
}
